machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,917 @@
+	// +kubebuilder:validation:Required
+	Images map[string]string `json:"images"`
+
+	// BaseOSContainerImage is the new-format container image for operating system updates.
+	// +kubebuilder:validation:Required
+	BaseOSContainerImage string `json:"baseOSContainerImage"`
+
+	// BaseOSExtensionsContainerImage is the matching extensions container for the new-format container
+	// +optional
+	BaseOSExtensionsContainerImage string `json:"baseOSExtensionsContainerImage"`
+
+	// OSImageURL is the old-format container image that contains the OS update payload.
+	// +optional
+	OSImageURL string `json:"osImageURL"`
+
+	// releaseImage is the image used when installing the cluster
+	// +kubebuilder:validation:Required
+	ReleaseImage string `json:"releaseImage"`
+
+	// proxy holds the current proxy configuration for the nodes
+	// +kubebuilder:validation:Required
+	// +nullable
+	Proxy *configv1.ProxyStatus `json:"proxy"`
+
+	// infra holds the infrastructure details
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:Required
+	// +nullable
+	Infra *configv1.Infrastructure `json:"infra"`
+
+	// dns holds the cluster dns details
+	// +kubebuilder:validation:EmbeddedResource
+	// +kubebuilder:validation:Required
+	// +nullable
+	DNS *configv1.DNS `json:"dns"`
+
+	// ipFamilies indicates the IP families in use by the cluster network
+	// +kubebuilder:validation:Required
+	IPFamilies IPFamiliesType `json:"ipFamilies"`
+
+	// networkType holds the type of network the cluster is using
+	// XXX: this is temporary and will be dropped as soon as possible in favor of a better support
+	// to start network related services the proper way.
+	// Nobody is also changing this once the cluster is up and running the first time, so, disallow
+	// regeneration if this changes.
+	// +optional
+	NetworkType string `json:"networkType,omitempty"`
+
+	// Network contains additional network related information
+	// +kubebuilder:validation:Required
+	// +nullable
+	Network *NetworkInfo `json:"network"`
+}
+
+// ImageRegustryBundle contains information for writing image registry certificates
+type ImageRegistryBundle struct {
+	// file holds the name of the file where the bundle will be written to disk
+	// +kubebuilder:validation:Required
+	File string `json:"file"`
+	// data holds the contents of the bundle that will be written to the file location
+	// +kubebuilder:validation:Required
+	Data []byte `json:"data"`
+}
+
+// IPFamiliesType indicates whether the cluster network is IPv4-only, IPv6-only, or dual-stack
+type IPFamiliesType string
+
+const (
+	IPFamiliesIPv4                 IPFamiliesType = "IPv4"
+	IPFamiliesIPv6                 IPFamiliesType = "IPv6"
+	IPFamiliesDualStack            IPFamiliesType = "DualStack"
+	IPFamiliesDualStackIPv6Primary IPFamiliesType = "DualStackIPv6Primary"
+)
+
+// Network contains network related configuration
+type NetworkInfo struct {
```

Comment:
Godocs talking about `Network` but the type name is `NetworkInfo`.  Not clear to me what `Info` adds to the name; all of these structures are designed to carry information.  But a common pattern for `*Spec` child types is to be `*Spec` themselves, so `ControllerConfigSpec` would have `NetworkSpec` or, if you wanted to namespace defensively, `ControllerConfigNetworkSpec`.

Reply:
Moving [here][1]; feel free to mark this thread resolved.

[1]: https://github.com/openshift/api/pull/1453#discussion_r1337873643